Add remaining parts of the Veeam content pack

nikita-konoplin opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

ssokolovich commented on Packs/Veeam/Integrations/VeeamONERESTAPI/VeeamONERESTAPI_description.md:
Please add more information about how to configure this integration. 
For example:
* Add reference on how to obtain the API key
* Which permissions are required for his integration
* What are the default settings that need to be added for classifier/mapper/incident type?

As more information will be presented here, it will be easier for the user to decide how to proceed with the configuration flow.
![image](https://github.com/user-attachments/assets/1619c268-6576-48bd-8990-4acd800805b7)

nikita-konoplin replied:
Additional information added.
![Screenshot 2024-08-07 142614](https://github.com/user-attachments/assets/5b2e7657-ab6e-4b12-b872-b9d39b8d4cfb)

ssokolovich commented on Packs/Veeam/Playbooks/Veeam_Start_Instant_VM_Recovery_Automatically.yml:
```diff
@@ -0,0 +1,1180 @@
+      iscommand: false
+      max: 100
+      wait: 1
+    nexttasks:
```
Please connect this task to `Done` or other task as needed.

nikita-konoplin replied:
Done

ssokolovich reviewed (commented):
@nikita-konoplin - looks good! 🎉 
A couple of notes from my side:
1. Would like to see a video of a working dashboard ( hard for me to test that since I don't have Veeam incidents on my machines). 🙏 
2. Please check if the `"unsearchable"` is set to True on all the new fields you've contributed to us. 
3. Missing classifier/mapper here - so the data from injected incidents will be mapped to system fields (also for those that you've contributed here).
4. Strongly suggest going over our OOTB incident fields and trying to avoid creating new ones. I've added some examples but assuming we can find suitable replacements for those fields you've created.
5. Using extraction on ALL is a bad practice from our side, as it may impact server performances. Please choose associated incident fields that are related to this incident type (per your contributed mapper).
6. Saw that you've used the playbook `"playbook0"` in the incident types. Please note that his PB is not in your contribution or our repo. Assuming it was used for tests - so pls change that accordingly.
7. Add more information to the integration description. It will benefit the user once he needs to configure the integration.
8. Missing the "Veeam Rest API"  so I can see the playbook flow properly, can you please add that to this PR? Or guide me on how to obtain that integration?


That will be the first assessment from my end. Assuming due to those comments you might change the PR and tweak it a bit. Once you've done - please re-request my review.
Cheers!

nikita-konoplin commented on Packs/Veeam/IncidentTypes/incidenttype-Incremental_Backup_Size.json:
```diff
@@ -0,0 +1,28 @@
+    "disabled": false,
+    "extractSettings": {
+        "fieldCliNameToExtractSettings": {},
+        "mode": "All"
```
done

nikita-konoplin commented on Packs/Veeam/IncidentTypes/incidenttype-Incremental_Backup_Size.json:
```diff
@@ -0,0 +1,28 @@
+    "locked": false,
+    "name": "Incremental Backup Size",
+    "onChangeRepAlg": 1,
+    "playbookId": "playbook0",
```
This was the "Default" playbook. We removed it.

melamedbn replied:
Just confirm that intended behaviour is no default playbook at all.

nikita-konoplin commented on Packs/Veeam/IncidentFields/incidentfield-Event_Created_By.json:
```diff
@@ -0,0 +1,27 @@
+{
+    "associatedToAll": true,
+    "caseInsensitive": true,
+    "cliName": "eventcreatedby",
```
done

nikita-konoplin commented:
Hello @ssokolovich ,
Thank you for PR review!
 
We have fixed all the comments,
but we also have a few questions:
 1. Would like to see a video of a working dashboard.
 - How to securely share the video? I can share it Slack in private message if needed.
 4. Strongly suggest going over our OOTB incident fields and trying to avoid creating new ones. I've added some examples but assuming we can find suitable replacements for those fields you've created.
 - We found only a few suitable fields (sourcehostname, sourcecreatedby, sourcecategory, sourceid, alerted, numberoffoundrelatedalerts). But we still need to create 26 new fields. Is it ok?
7. Missing the "Veeam Rest API" so I can see the playbook flow properly, can you please add that to this PR? Or guide me on how to obtain that integration?
- @edik24  told us to create separate PR for one integration, and the rest in another one. So here it is: [Add integration for Veeam Backup & Replication REST API by nikita-konoplin · Pull Request #35670 · demisto/content (](https://github.com/demisto/content/pull/35670)[github.com](http://github.com/)[)](https://github.com/demisto/content/pull/35670)

@edik24 @samuelFain Since Sasha is on PTO, can someone take a look at our changes?

nikita-konoplin commented on Packs/Veeam/Dashboards/dashboard-Veeam_Incident_Dashboard.json:
How to securely share the video? I can share it Slack in private message if needed.

melamedbn replied:
Hi @nikita-konoplin,

Please send it to me via Slack DFIR channel.

Thanks.

nikita-konoplin replied:
@melamedbn video sent

samuelFain commented on Packs/Veeam/Integrations/VeeamONERESTAPI/VeeamONERESTAPI.py:
```diff
@@ -0,0 +1,629 @@
+    )
+    assert first_fetch_time
+
+    global http_request_timeout_sec
```
Please avoid initializing global arguments. Initialize as a regular argument in main and send it as function argument to the relevant commands.

nikita-konoplin replied:
done

samuelFain commented on Packs/Veeam/Integrations/VeeamONERESTAPI/VeeamONERESTAPI.py:
```diff
@@ -0,0 +1,629 @@
+        arg_name='First fetch time',
+        required=True
+    )
+    assert first_fetch_time
```
If `arg_to_datetime` returns `None` the integration will break.
Please add a more graceful handling here and insure that a valid default value will be set in case of a `None` response, Something like:
```
if not first_fetch_datetime:
    first_fetch_time = datetime.now()
```

nikita-konoplin replied:
done

samuelFain commented on Packs/Veeam/Integrations/VeeamONERESTAPI/VeeamONERESTAPI_image.png:
Please add relevant integration image.
See image requirements [here](https://xsoar.pan.dev/docs/integrations/integration-logo).

nikita-konoplin replied:
Logo images are not ready yet, we will attach them when ready.

nikita-konoplin replied:
Author and integration images were added

samuelFain reviewed (changes requested):
@nikita-konoplin Looks good!
Please see my comments.

Also, please merge your base branch from content/master, and then pull from your base branch to this branch. This is needed due to a change made in environment configuration files of content to allow all the Github actions checks to run in this PR.

Thanks!

nikita-konoplin commented:
@samuelFain Thank you!
Logo images are not ready yet, we will attach them when ready.

melamedbn commented on Packs/Veeam/IncidentFields/incidentfield-Hostname.json:
```diff
@@ -0,0 +1,27 @@
+{
+    "associatedToAll": true,
+    "caseInsensitive": true,
+    "cliName": "hostname",
```
1. Affected Machine can be mapped to Affected Hosts
2. Alarm status can be mapped to Source or External Status
3. Event Source/Type Description can be mapped to Event Description

Besides that, any other field should be prefixed with 'Veeam' e.g. 
1. Veeam Remediation
2. Veeam Backup...
etc.

nikita-konoplin replied:
@melamedbn done

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=35706) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=35706) before we can accept your contribution.<br/>**3** out of **5** committers have signed the CLA.<br/><br/>:white_check_mark: yuvalbenshalom<br/>:white_check_mark: nikita-konoplin<br/>:white_check_mark: yasta5<br/>:x: YaelShamai<br/>:x: itssapir<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=35706) it.</sub>

samuelFain commented:
Closing this PR as it was integrated into https://github.com/demisto/content/pull/36097.
